packages/shared/src/models-hdf5.ts

```diff
@@ -157,6 +161,8 @@ export type Primitive<T extends DType> = T extends NumericType
   ? boolean
   : T extends ComplexType
   ? H5WebComplex
+  : T extends PrintableCompoundType
+  ? Primitive<PrintableType>[]
```

Comment:
This is where having a subtype `PrintableCompoundType` is useful. We can then infer the `Primitive` of this dtype and therefore the `Value` type when fetching the value in `ValueFetcher`.

The values of compound is an array containing tuple of the field entries. For example, for the compound dataset `compound/dogs` in https://h5web.panosc.eu/h5grove?file=grove.h5 where we have three fields (`name`, `age` and `weight`) the value is
```
[
  ['Rex', 9, 81],
  ['Fido', 3, 27],
]
```

The nice thing about `PrintableType` is that we don't really need to know what is the exact tuple type (here  `[string, number, number]`) as long as the tuple contains printable types which is exactly what `Primitive<PrintableType>[]` describes.

Reply:
This is great. 🎉

May I just suggest adding a generic to `CompoundType`?

```ts
export interface CompoundType<T = DType> {
  class: DTypeClass.Compound;
  fields: Record<string, T>;
}

// Then you can use `CompoundType<PrintableType>` directly or make an alias if you prefer:
export type PrintableCompoundType = CompoundType<PrintableType>
```